### DISCUSSION TITLE
Deploy operator in multiple namespaces

### DISCUSSION THREAD

**Discussion by popdaniel942** [strimzi/strimzi-kafka-operator#6646] [Q&A]
Hello.
I've tried deploying the operator with Helm in multiple namespaces and the only way I managed to do it was by creating global resources in the first deployment and without creating them in the second deployment. The problem seems to be because of ClusterRoles and ClusterRoleBindings because they would have the same name if they are created in 2 deployments. Can't they be changed so that they have the Release.Name in front of all of their names or something like that so each deployment would have their own ClusterRoles and ClusterRoleBindings?

Thanks in advance!

**Comment by scholzj** [comment]
Kubernetes is not multi-tenant. So running multiple separate instances of the operator will be always tricky. In this case, the ClusterRoles need to stay and you need to normally use the ClusterRoles from the newest operator version you use. As for the ClusterRoleBindings, their names to not matter, so you can just rename them as you wish.

The reason why they are named with version is that it would make it harder to delete them during upgrades and it wouldn't really solcve the main issues in running multiple operators in parallel.

**Reply by popdaniel942** [reply]
But why do the names of the ClusterRoles matter? We want to deploy in multiple namespaces while creating global resources in all of them, but now we can't because the names are hardcoded and they clash with each other. If we just deploy the global resources in one namespace and then we uninstall that deployment, the global resources will be deleted and the operator won't work anymore in the other namespaces.

**Reply by scholzj** [reply]
The ClusterRoles are needed for the operator and for the operands -> they are referred to by dynamically created ClusterRoleBindings. At leats some of them -> for those used only for the cluster operator the name does not really matter and you can rename it as well.
